### PR TITLE
Fix matching test app (only sift)

### PR DESCRIPTION
**Pull request by nmoehrle**
Maybe those testing apps should be included in Travis elsewise they will break whenever somebody refactors something.

### CODE REVIEW

File: libs/sfm/_test_matching.cc [hunk 1]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
 }
 
 void
+convert_sift_descriptors(sfm::Sift::Descriptors const & img_descr,
+    util::AlignedMemory<math::Vec128f> * descr)
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: no space before `*`

File: libs/sfm/_test_matching.cc [hunk 2]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
 }
 
 void
+convert_sift_descriptors(sfm::Sift::Descriptors const & img_descr,
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: no space before `&`

File: libs/sfm/_test_matching.cc [hunk 3]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
...
+	convert_sift_descriptors(feat2.sift_descriptors, &sift_descr2);
+
+    sfm::Matching::Result sift_matching;
+    sfm::Matching::twoway_match(sift_matching_opts,
+        sift_descr1.data()->begin(), sift_descr1.size(),
+        sift_descr2.data()->begin(), sift_descr2.size(),
+        &sift_matching);
+    sfm::Matching::remove_inconsistent_matches(&sift_matching);
+
+    sfm::Matching::Result const & matching = sift_matching;
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: no space before `&`

File: libs/sfm/_test_matching.cc [hunk 4]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
...
+    sfm::FeatureSet feat1(feature_set_opts);
     feat1.compute_features(image1);
-    sfm::FeatureSet feat2(opts);
+    sfm::FeatureSet feat2(feature_set_opts);
     feat2.compute_features(image2);
 
-    sfm::Matching::Result matching;
-    feat1.match(feat2, &matching);
+    util::AlignedMemory<math::Vec128f, 16> sift_descr1, sift_descr2;
+	convert_sift_descriptors(feat1.sift_descriptors, &sift_descr1);
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: indent broken

File: libs/sfm/_test_matching.cc [hunk 5]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
...
+	convert_sift_descriptors(feat2.sift_descriptors, &sift_descr2);
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: Every time you insert a tab, a little kitten dies.

File: libs/sfm/_test_matching.cc [hunk 6]
```diff
@@ -48,26 +48,49 @@ visualize_matching (sfm::Matching::Result const& matching,
...
 
 void
+convert_sift_descriptors(sfm::Sift::Descriptors const & img_descr,
+    util::AlignedMemory<math::Vec128f> * descr)
+{
+    descr->resize(img_descr.size());
+    float* data_ptr = descr->data()->begin();
+    for (std::size_t i = 0; i < img_descr.size(); ++i, data_ptr += 128)
+    {
+        sfm::Sift::Descriptor const & d = img_descr[i];
```
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: no space before `&`

File: libs/sfm/_test_matching.cc (hunk 2 above)
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: call the argument `input` or `sift_descr`

File: libs/sfm/_test_matching.cc (hunk 1 above)
Comment (simonfuhrmann) [inline libs/sfm/_test_matching.cc]: call the argument `output` or `aligned_descr`

**Comment by simonfuhrmann** [comment]
Maybe they should be built in Travis. On the other hand, these are just for testing during development, so it's not that critical...

File: libs/sfm/_test_matching.cc (hunk 5 above)
Comment (nmoehrle) [inline libs/sfm/_test_matching.cc]: I have honestly no idea how I managed to insert a tab - vim is configured correctly

**Pull request merged**